cinder/volume/drivers/dell_emc/unity/client.py

```diff
@@ -111,6 +111,8 @@ def delete_lun(self, lun_id):
         def _delete_lun_if_exist(force_snap_delete=False):
             """Deletes LUN, skip if it doesn't exist."""
             try:
+                if hasattr(lun, 'host_access') and lun.host_access:
```

Comment:
Looks like it will delete the LUN anyway. 
Do we provide a force option to delete LUN with host access? But if no force enabled, the LUN could not be deleted.

Reply:
There is a `force` option in the OpenStack/Cinder command. But this option is handled in the framework layer, not passed into the driver. So, the driver doesn't have any information about the volume force deletion.
I requested for passing the force from the framework to the driver. The community didn't buy-in. They thought the driver should always delete the volume if it is told to.

@yong-huang , I think we could add some INFO logs here in case we need them to triage the issue.

Reply:
`force` option is not used to delete the attached volume, it is used to delete volumes in error status, such as 'error', 'error_restoring', 'error_extending' or 'error_managing'.
When deleting volume,  if it is attached, framework will raise VolumeAttached exception. The `force` option will not pass to driver.
One solution is to add an option in driver to control whether to delete the volumes with hosts accessed, if the option is set to False, will keep the old logic and not remove the host from lun.

Reply:
Adding an option in the driver is better, so that the customer is aware that the LUN would be delete with host access.

Reply:
I think the community won't accept the option either. Check this review: https://review.opendev.org/#/c/611515/. I got two -2 from two core reviewers who thought it's not necessary to add an option to force delete a LUN with snapshots.

Reply:
Is there any way to let the customer know the LUN would be deleted even with host access?

Reply:
> Is there any way to let the customer know the LUN would be deleted even with host access?

Maybe the only way is logging.

Reply:
The user may not look at the logging unless it shows on the screen. 
All right. I don't have other comments if adding an option to the driver is not allowed by OpenStack community.

Reply:
As Yong stated previously, from OpenStack view, if a volume is in-use, it will raise VolumeAttached exception when the user tries to delete it. So, the deletion can be passed to the driver only if OpenStack sees the volume is not attached. If Unity still has the volume attached, it could be an error that occurred in Unity (why Unity didn't detach the volume successfully last time). In this case, OpenStack always wants this volume to be detached first and then deleted (the option seems useless) because the volume is not in-use any more from its view.

Reply:
Got it, that makes sense.

Reply:
> There is a `force` option in the OpenStack/Cinder command. But this option is handled in the framework layer, not passed into the driver. So, the driver doesn't have any information about the volume force deletion.
> I requested for passing the force from the framework to the driver. The community didn't buy-in. They thought the driver should always delete the volume if it is told to.
> 
> @yong-huang , I think we could add some INFO logs here in case we need them to triage the issue.

Done. INFO logs added.